purchase_order_general_discount/models/purchase_order.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from lxml import etree
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    general_discount = fields.Float(
+        digits=dp.get_precision('Discount'),
+        string='General Discount (%)',
+    )
+
+    _sql_constraints = [
+        ('general_discount_limit', 'CHECK (general_discount <= 100.0)',
+         'Discount must be lower than 100%.'),
+    ]
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        super().onchange_partner_id()
+        self.general_discount = self.partner_id.purchase_general_discount
+
+    def _get_general_discount_field(self):
+        """We can set in settings another discount field to be applied
+           For example, if we had purchase_triple_dicount, we could set the
+           general discount in discount3 to be applied after all other
+           discounts"""
+        discount_field = self.company_id.purchase_general_discount_field
+        return discount_field or 'discount'
+
+    @api.onchange('general_discount')
+    def onchange_general_discount(self):
+        discount_field = self._get_general_discount_field()
+        self.mapped('order_line').update({
+            discount_field: self.general_discount,
+        })
+
+    def action_update_general_discount(self):
```

Comment:
Why do you need this?

Reply:
I use it as wrapper to trigger it from ui